docs/usage/working-with-bundlers.md

```diff
@@ -10,50 +10,44 @@ project.
 ## Vite
 
 ```{note}
-The following instructions have been tested with Pyodide 0.26.0 and Vite 5.2.13.
+The following instructions have been tested with Pyodide 0.26.2, Vite 5.4.9, and
+vite-plugin-pyodide 2.0.0.
 ```
 
-First, install the Pyodide npm package:
+First, install the Pyodide and vite-plugin-pyodide npm packages:
 
 ```
-$ npm install pyodide
+$ npm install pyodide vite-plugin-static-copy
 ```
 
-Then, in your `vite.config.js` file, exclude Pyodide from [Vite's dependency
+Then, in your `vite.config.mjs` file, exclude Pyodide from [Vite's dependency
 pre-bundling][optimizedeps] by setting `optimizeDeps.exclude` and ensure that
 all Pyodide files will be available in `dist/assets` for production builds by
 using a Vite plugin:
 
 ```js
 import { defineConfig } from "vite";
-import { copyFile, mkdir } from "fs/promises";
+import { viteStaticCopy } from "vite-plugin-static-copy";
 import { dirname, join } from "path";
 import { fileURLToPath } from "url";
 
+const PYODIDE_EXCLUDE = ["!**/package.json", "!**/*.{md,html}", "!**/*.d.ts"];
+
+export function viteStaticCopyPyodide() {
+  const pyodideDir = dirname(fileURLToPath(import.meta.resolve("pyodide")));
+  return viteStaticCopy({
+    targets: [
+      {
+        src: [join(pyodideDir, "*")].concat(PYODIDE_EXCLUDE),
+        dest: "assets",
```

Comment:
Will this include all the wheels in the directory? Doesn't it explode the final output size?

Reply:
No, because the NPM package does not all the wheels.

`npm install pyodide` will install the NPM package and
```
pyodideDir = dirname(fileURLToPath(import.meta.resolve("pyodide")))
```
will find that NPM package.

Unless the user has done something really strange like make the full redistribution with all the wheel look like an NPM packege under some `node_modules` directory. I can't think of any other corner case off the top of my head.

Reply:
The "NPM package" being the locally stored package downloaded from NPM.

Reply:
I see. That makes sense.

However, when you run Pyodide in local Node.js environment, it has a side effect of downloading packages into the `node-modules/pyodide` directory on demand as a side effect, which means that the content inside the node-modules can change. So I am a little worried that people might be confused about it.

Reply:
Ohhhhh, yeah, good point! Maybe I should add another exclusion pattern. Strange that I didn't suffer from this. Now that you point this out I'm surprised this hasn't happened to me already.

Reply:
I see, when I was testing this example code it was in a nice clean environment where nobody had started to party in the local `node-modules/pyodide` directory.

I also see now in a dirtier project that prettier has also been partying in this directory has created a `node_modules/pyodide/node_modules/.cache/prettier/` subdirectory.

Hmmm, maybe the inclusion list approach is safer than an exclusion list approach. :face_with_diagonal_mouth:

Reply:
I'm thinking the trade-off depends on the respective risks of:

1) some unexpected and unpredictable files and folders appearing in the local `node-modules/pyodide` directory
VS
2) a Pyodide upgraded version changing the list of pyodide files that need to be redistributed in the build (and thus code copied from an older example longer working).

Reply:
> a Pyodide upgraded version changing the list of pyodide files that need to be redistributed in the build (and thus code copied from an older example suddenly no longer working).

Yes, we do have this risk. So in [webpack-plugin](https://github.com/pyodide/pyodide-webpack-plugin/blob/main/lib/patterns.ts) we maintain a list of "core" files in the Pyodide distribution.

Reply:
Oh, great data point to see what webpack-plugin is doing. It looks like it WAS taking the inclusion approach for version up until 0.24.0 and has since taken an exclusion approach.
Interestingly, webpack-plugin seems to always INCLUDE package.json whereas I was assuming it should be excluded.

Reply:
I was starting to question the exclusion approach, but now that I see webpack-plugin ended up taking the exclusion approach, now I am back to thinking exclusion is the way to go. Let me know if you think we should discuss more whether exclusion approach is the right approach.

To wrap up this sub-thread here, we've realized two additional types of files that should be excluded:

1. all wheels "**/*.whl" and
2. this weird case I see locally of some files appearing under "node_modules" *within* the pyodide directory, so everything that might be there should be eliminated too.

I'll add another commit for these.